Binary/Loader/FileSystemLoader.php

```diff
@@ -38,30 +38,48 @@ class FileSystemLoader implements LoaderInterface
     /**
      * @param MimeTypeGuesserInterface  $mimeGuesser
      * @param ExtensionGuesserInterface $extensionGuesser
-     * @param string[]                  $dataRoots
-     * @param LocatorInterface          $locator
+     * @param string[]|LocatorInterface $locator
      */
```

Comment:
Hmm; while I understand your intention, `@deprecated` is a doc-block scope declaration that would cause IDEs and other tools to believe the entire `__construct()` method had been deprecated, which is obviously not the case. Perhaps a comment explaining the depreciation would be more appropriate? We don't want people seeing "FileSystemLoader.php::~~__construct()~~" in their IDEs.